spec/inputs/xmpp_spec.rb

```diff
@@ -1 +1,106 @@
+# encoding: utf-8
 require "logstash/devutils/rspec/spec_helper"
+require 'logstash/inputs/xmpp'
+
+module Jabber
```

Comment:
I considered this and made a point about it in the Reviewers Notes above.  What rule of thumb should we apply here? On the one hand it is tidy to move the mocks to their own file and on the other hand, moving them will make it less easy to read and follow the whole spec suite in one go.

I cannot find any solid guidance in the Internet about extract or not wrt helpers.  Even the [Rspec style guide](http://betterspecs.org/) is silent on this.

The only RSpec convention is to put them in /support and they get auto-loaded but then the same people that don't know what described_class means will not know about this RSpec convention either.